helm/examples/azure/hpcc-azurefile/templates/_helpers.tpl

```diff
@@ -61,3 +61,82 @@ Create the name of the service account to use
     {{ default "default" .Values.serviceAccount.name }}
 {{- end -}}
 {{- end -}}
+
+{{/*
+Add common mountOptions
+*/}}
+{{- define "hpcc-azurefile.addCommonMountOptions" }}
+mountOptions:
+- uid=10000 # uid of user 'hpcc'
+- gid=10001 # gid of group 'hpcc'
+- dir_mode=0700 # user read/write/execute
+- file_mode=0600 # user read/write
+{{- if .root.Values.common.shareName }}
+- mfsymlinks
+- nobrl
+{{- end -}}
+{{- end }}
+
+{{/*
+Create PersistentVolume
+*/}}
+{{- define "hpcc-azurefile.addStoragePV" -}}
+{{- $plane := .plane -}}
+{{- $common := .root.Values.common -}}
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: {{ printf "%s-%s-pv" $plane.name (include "hpcc-azurefile.fullname" .root) }}
+  labels:
+    store: {{ printf "%s-%s-pv" $plane.name (include "hpcc-azurefile.fullname" .root) }}
+spec:
+  capacity:
+    storage: {{ $plane.size }}
+  accessModes:
+    - {{ $plane.rwmany | default false | ternary "ReadWriteMany" "ReadWriteOnce" }}
+  azureFile:
+    secretName: {{ $common.secretName }}
+    secretNamespace: {{ $common.secretNamespace }}
+    shareName: {{ $plane.shareName }}
+    readOnly: false
+  {{- include "hpcc-azurefile.addCommonMountOptions" . | indent 2}}
+{{- end -}}
+
+{{/*
+Create StorageClass name
+*/}}
+{{- define "hpcc-azurefile.SCName" -}}
+{{- $plane := .plane -}}
+{{- if not $plane.shareName -}}
+{{ printf "%s-%s-sc" $plane.name (include "hpcc-azurefile.fullname" .root) | trunc 63 }}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Create StorageClass
+*/}}
+{{- define "hpcc-azurefile.addStorageSC" -}}
+{{- $plane := .plane -}}
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: {{ printf "%s-%s-sc" $plane.name (include "hpcc-azurefile.fullname" .root) }}
+provisioner: kubernetes.io/azure-file
+{{- include  "hpcc-azurefile.addCommonMountOptions" . }}
+parameters:
+  skuName: {{ $plane.sku | default "Standard_LRS" }}
+reclaimPolicy: Retain
+allowVolumeExpansion: true
+{{- end -}}
+
+{{/*
+Add PV selector
+*/}}
+{{- define "hpcc-azurefile.addPVSelector" -}}
+{{- $plane := .plane -}}
+{{- if .root.Values.common.shareName -}}
+selector:
+  matchLabels:
+    store: {{ printf "%s-%s-pv" $plane.name (include "hpcc-azurefile.fullname" .root) }}
```

Comment:
can this use "hpcc-azurefile.SCName" so common?

Reply:
storageClass can and I replace it.  I add a "{{- define "hpcc-azurefile.PVName" -}}" to make PV name definition in one place.